File: MicroCli/src/main/resources/micronaut/security/api key/java/JDBC/token/ApiKeyTokenReader.java
package ${securityPackage}.token;


import io.micronaut.security.token.reader.HttpHeaderTokenReader;
import jakarta.inject.Singleton;

@Singleton
public class ApiKeyTokenReader extends HttpHeaderTokenReader {
    private static final String API_TOKEN_HEADER = "${securityTokenHeader}";

    @Override
    protected String getPrefix() {
        return null;
    }

    @Override
    protected String getHeaderName() {
        return API_TOKEN_HEADER;
    }
}

File: MicroCli-deprecated/src/main/resources/micronaut/security/api key/kotlin/JDBC/token/ApiKeyTokenReader.kt
package ${securityPackage}.token

import io.micronaut.security.token.reader.HttpHeaderTokenReader
import jakarta.inject.Singleton

@Singleton
class ApiKeyTokenReader : HttpHeaderTokenReader() {
    protected val prefix: String?
        protected get() = null

    companion object {
        protected val headerName = "API-HEADER"
            protected get() = Companion.field
    }
}